src/auto-evo/mutation_strategy/UpgradeOrganelle.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Frozen;
+using System.Collections.Generic;
+using System.Linq;
+using AutoEvo;
+
+internal class UpgradeOrganelle : IMutationStrategy<MicrobeSpecies>
+{
+    private readonly FrozenSet<OrganelleDefinition> allOrganelles;
+    private IComponentSpecificUpgrades upgrade;
+
+    public UpgradeOrganelle(Func<OrganelleDefinition, bool> criteria, IComponentSpecificUpgrades upgrade)
+    {
+        allOrganelles = SimulationParameters.Instance.GetAllOrganelles().Where(criteria).ToFrozenSet();
+        this.upgrade = upgrade;
+    }
+
+    public bool Repeatable => true;
+
+    public List<Tuple<MicrobeSpecies, float>>? MutationsOf(MicrobeSpecies baseSpecies, float mp, bool lawk,
+        Random random)
+    {
+        if (allOrganelles.Count == 0)
+        {
+            return null;
+        }
+
+        var mutated = new List<Tuple<MicrobeSpecies, float>>();
+
+        MicrobeSpecies newSpecies = (MicrobeSpecies)baseSpecies.Clone();
+
+        foreach (OrganelleTemplate organelle in newSpecies.Organelles.Where(x => allOrganelles.Contains(x.Definition)))
+        {
+            organelle.Upgrades ??= new OrganelleUpgrades();
+            organelle.Upgrades.CustomUpgradeData = upgrade;
```

Comment:
I was going to comment that it would be slightly more efficient to not clone the custom upgrade data here, but that would be quite longer code and require adding new configuration for the organelle clone method to not clone the upgrades. As all organelles are cloned anyway this probably doesn't matter so I won't ask you to do that unless you reall want to.